WSFrontend/org.openl.rules.ruleservice.ws/src/org/openl/rules/ruleservice/storelogdata/CacheAndWriteOutputStream.java

```diff
@@ -0,0 +1,79 @@
+package org.openl.rules.ruleservice.storelogdata;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.cxf.io.CachedOutputStream;
+
+public class CacheAndWriteOutputStream extends CachedOutputStream {
+
+    OutputStream flowThroughStream;
+    ByteArrayOutputStream flowThroughStreamCache = new ByteArrayOutputStream();
+    long count;
+    long limit = Long.MAX_VALUE;
+
+    public CacheAndWriteOutputStream(OutputStream stream) {
+        super();
+        if (stream == null) {
+            throw new IllegalArgumentException("Stream may not be null");
+        }
+        flowThroughStream = stream;
+    }
+
+    public void setCacheLimit(long l) {
+        limit = l;
+    }
+
+    public void closeFlowthroughStream() throws IOException {
+        flowThroughStream.flush();
+        flowThroughStream.close();
+    }

```

Comment:
I cant find where this method is used, so it makes sense to override close() methods to close all streams
```suggestion
    @Override
    public void close() throws IOException {
        super.close();
        flowThroughStream.flush();
        flowThroughStream.close();
    }
```